src/main/scala/molmed/apps/MakeSolidProjectHierarchy.scala

```diff
@@ -0,0 +1,322 @@
+package molmed.apps
+
+import java.io.File
+import scala.io.Source
+import molmed.utils.GeneralUtils
+import java.io.PrintWriter
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.FileAlreadyExistsException
+import scopt._
+import java.io.FileWriter
+
+/**
+ * Utility program to convert sthlm sequencing platform meta format to UU SNP format.
+ */
+object MakeSolidProjectHierarchy extends App {
+
+  case class Config(
+    sthlmRoot: Option[File] = None,
+    newUppsalaStyleRoot: Option[File] = None,
+    flowcells: Seq[String] = Seq())
+
+  val parser = new OptionParser[Config]("MakeSolidProjectHierarchy$") {
+    head("MakeSolidProjectHierarchy$", " - A utility program to convert a sthlm style project dir to a ua style one. \n" +
+      "Run example: " +
+      "./MakeSolidProjectHierarchy$ " +
+      " --input_root <sthlm project root folder> " +
+      " --out_root <root of uppsala style project>")
+
+    opt[File]('i', "input_root") required () valueName ("A SOLiD style project directory.") action { (x, c) =>
+      c.copy(sthlmRoot = Some(x))
+    } text ("This is a required argument.")
+
+    opt[File]('o', "out_root") required () valueName ("Root dir where the ua style directories will be written.") action { (x, c) =>
+      c.copy(newUppsalaStyleRoot = Some(x))
+    } text ("This is a required argument.")
+
+    opt[String]('f', "flowcell") unbounded () optional () valueName ("Data format conversion will be restricted to the following flowcells.") action { (x, c) =>
+      c.copy(flowcells = c.flowcells :+ x)
+    } text ("This is a optional argument, and it can be specified multiple times.")
+
+  }
+
+  // Start up the app!
+  parser.parse(args, new Config()) map { config =>
+
+    runApp(config)
+
+  } getOrElse {
+    // arguments are bad, usage message will have been displayed
+  }
+
+  /**
+   * Utility functions
+   */
+
+  /**
+   * List all subdirectories of dir
+   */
+  def listSubDirectories(dir: File): Seq[File] = {
```

Comment:
Yes, I first started with duplicating the file/class closest to the one I want to write and start making changes therein. It is a lazy and error-prone process, but I since any file with name containing "Solid" is ONLY to deal with colorspace data. So the risk is only on my shoulder, they do not affect the upstream code. With that said, this adaptation work is still going on and I do improve them quickly.